boot/zephyr/main.c

```diff
@@ -175,6 +183,40 @@ static void do_boot(struct boot_rsp *rsp)
 }
 #endif
 
+#if defined(CONFIG_LOG) && !defined(CONFIG_LOG_IMMEDIATE) &&\
+    !defined(CONFIG_LOG_PROCESS_THREAD)
+#include <logging/log_ctrl.h>
+
+/* most simple log processing theread */
+void boot_log_therad_func(void *dummy1, void *dummy2, void *dummy3)
+{
+    (void)dummy1;
+    (void)dummy2;
+    (void)dummy3;
+
+     log_init();
+
+     while (1) {
+             if (log_process(false) == false) {
+                    k_sleep(30);
+             }
+     }
+}
+
+K_THREAD_STACK_DEFINE(boot_log_stack, 768);
```

Comment:
Aren't stack sizes of threads typically user configurable? In this case I kinda agree that it can be hardcoded, because its logging, and will never be in released code (AFAIK), you could consider if you want to add some kconfig option for this though.